src/test_typing_extensions.py

```diff
@@ -4177,6 +4177,121 @@ class AllTheThings(TypedDict):
         self.assertEqual(AllTheThings.__optional_keys__, frozenset({'c', 'd'}))
         self.assertEqual(AllTheThings.__readonly_keys__, frozenset({'a', 'b', 'c'}))
         self.assertEqual(AllTheThings.__mutable_keys__, frozenset({'d'}))
+    
+    def test_extra_keys_non_readonly(self):
+        class Base(TypedDict, closed=True):
+            __extra_items__: str
+        
+        class Child(Base):
+            a: NotRequired[int]
+        
+        self.assertEqual(Child.__required_keys__, frozenset({}))
```

Comment:
Also test the value of `__closed__` and `__extra_items__`. Should also do this on most other classes below.